### PR TITLE
Iox #409 axivion analysis on CI

### PR DESCRIPTION
**Pull request by dkroenke**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md

## Notes for Reviewer
This PR brings initial CI tooling for scanning iceoryx with axivion 7.1.3 on GitHub. The current scan happens only on the master branch. After this PR is merged the scan will be triggered. The results are pushed into a dashboard on https://iceoryx-axivion.apex.ai.
Frequent contributors will get access to this dashboard to see the latest scan results on axivion.

More things to come (in progress):
- script for local scan
- scan pull-request branches

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- relates to #409

### CODE REVIEW

File: tools/axivion/axivion_config.json [hunk 1]
```diff
@@ -0,0 +1,1741 @@
...
+                "cmake build",
+                "AxivionLinker",
+                "Remove Build Directory",
+                "Unpin All Sources",
+                "AxivionCompiler"
+            ]
+        },
+        "Command": {
+            "_active": false,
+            "build_command": "./tools/iceoryx_build_test.sh examples dds-gateway clean",
```
Comment (mossmaurice) [inline tools/axivion/axivion_config.json:1638]: Shouldn't examples and dds-gateway be removed here?

**Review by mossmaurice — DISMISSED** [review]
LGTM, couple of questions came up

**Comment by mossmaurice** [comment]
After having a look at the scan results at https://iceoryx-axivion.apex.ai We shouldn't scan the C binding with the AUTOSAR C++ rules, right? Maybe @neillangmead41 can shed some light on this questions.

File: tools/axivion/axivion_config.json [hunk 2]
```diff
@@ -0,0 +1,1741 @@
+{
```
[the remark this answers is not in the document]
Comment (dkroenke) [reply tools/axivion/axivion_config.json:1]: Mapping table is work in progress and will come soon.

File: tools/axivion/axivion_config.json [hunk 3]
```diff
@@ -0,0 +1,1742 @@
...
+    },
+    "Project": {
+        "AxivionLinker": {
+            "_active": true,
+            "input_files": [
+                "$(ICEORYX_DIR)/build/install/prefix/bin/iox-cpp-publisher",
+                "$(ICEORYX_DIR)/build/install/prefix/bin/iox-cpp-subscriber",
+                "$(ICEORYX_DIR)/build/install/prefix/bin/iox-cpp-waitset-individual",
+                "$(ICEORYX_DIR)/build/install/prefix/bin/iox-roudi"
+            ],
```
Comment (mossmaurice) [inline tools/axivion/axivion_config.json:1620]: Why these selection of example here?

**Review by mossmaurice — APPROVED** [review]
LGTM

File: tools/axivion/axivion_config.json (hunk 3 above)
Comment (dkroenke) [reply tools/axivion/axivion_config.json:1620]: The examples are the entry point for the Axivion Linker to scan the iceoryx libraries. I put here the iox-roudi for all the roudi-related libraries and examples for pub/sub and waitset for the posh lib. Axivion will then later also cover the unused functionality of the libraries.

**Review by budrus — APPROVED** [review]

**Pull request merged**